### PR TITLE
Backend rich lo l api

### PR DESCRIPTION
**Pull request by RichCherng**
Just barebone logic, need clean up

### CODE REVIEW

File: src/LoLAPI/api.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import fs from 'fs';
+import request from 'request'
+import polyfill from 'babel-polyfill'
+
+let url_getStat = "https://na1.api.riotgames.com/lol/summoner/v3/summoners/by-name/"; // This need to be move out ot config file
+let url_matches = "https://na1.api.riotgames.com/lol/match/v3/matchlists/by-account/"
+let url_match = "https://na1.api.riotgames.com/lol/match/v3/timelines/by-match/"
+
+
+let key = fs.readFileSync('./api_key');
```
Comment (cheriejw) [inline src/LoLAPI/api.js]: create a config file to import the key directly rather than using fs

Comment (RichCherng) [reply src/LoLAPI/api.js]: yea, it's there for now. Need to create config file

File: src/LoLAPI/api.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+/*
+	This file contain ONLY LoL API and does not contain any logic for gathering data
+
+	Every function here is a call to LoL API only and should not do any complicated logi
+*/
+import fs from 'fs';
+import request from 'request'
+import polyfill from 'babel-polyfill'
+
+let url_getStat = "https://na1.api.riotgames.com/lol/summoner/v3/summoners/by-name/"; // This need to be move out ot config file
```
Comment (cheriejw) [inline src/LoLAPI/api.js]: Can use the URL_GETSTAT naming convention for constant values.

File: src/LoLAPI/api.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+let url_matches = "https://na1.api.riotgames.com/lol/match/v3/matchlists/by-account/"
+let url_match = "https://na1.api.riotgames.com/lol/match/v3/timelines/by-match/"
+
+
+let key = fs.readFileSync('./api_key');
+
+key = '?api_key=' + key;
+console.log(key.toString());
```
Comment (cheriejw) [inline src/LoLAPI/api.js]: clean up

File: src/LoLAPI/api.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+			});
+		} catch (err) {
+			console.log(" error request ", err);
+			process.exit();
+		}
+	})
+}
+
+const pi = 3.14159265359;
+export var calcCircumference = (radius) => {
```
Comment (cheriejw) [inline src/LoLAPI/api.js:96]: clean up

File: src/LoLAPI/api.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+				if (res && (res.statusCode === 200 || res.statusCode === 201)) {
+					// console.log(' response ', res.body.rates.INR);
+					resolve(res.body);
+				} else {
+					console.log('error body', body);
+					reject(new Error(`Error body: ${JSON.stringify(body)}`));
+				}
+			});
+		} catch (err) {
+			console.log(" error request ", err);
```
Comment (cheriejw) [inline src/LoLAPI/api.js]: change to '

File: src/LoLAPI/api.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+let url_match = "https://na1.api.riotgames.com/lol/match/v3/timelines/by-match/"
+
+
+let key = fs.readFileSync('./api_key');
+
+key = '?api_key=' + key;
+console.log(key.toString());
+
+
+export let getSummonerStat = async (_summonerName) => {
```
Comment (cheriejw) [inline src/LoLAPI/api.js:22]: Use 2 spaces as specified in design docs instead of tabs.

File: src/index.js [hunk 7]
```diff
@@ -1,6 +1,62 @@
-const express = require('express');
+// const express = require('express');
+// const app = express();
+
+import express from 'express';
+import {getMatch} from './LoLAPI/api.js'; 
+import {getSummoner} from './LoLAPI/summoner.js'
```
Comment (cheriejw) [inline src/index.js]: `import { getSummoner } from './dshfkasdjhf.js'`;

File: src/index.js [hunk 8]
```diff
@@ -1,6 +1,62 @@
...
+	});
+
+});
+
+
+app.get('/summoner/:summoner', (req, res) => {
+	getSummoner(req.params.summoner).then( (s) => {
+		res.json(s);
+	})
+})
```
Comment (cheriejw) [inline src/index.js]: ;

**Review by cheriejw — CHANGES_REQUESTED** [review]

**Comment by RichCherng** [comment]
Review POR FAAAAAVORRRR

**Comment by RichCherng** [comment]
BUT DON'T MERGE

**Review by cheriejw — CHANGES_REQUESTED** [review]
for .babelrc please use newline after the curly brackets and indent with 2 spaces the "presets": ["env"] like you did in the front-end babelrc file.

You don't have to, but would be nice to add comment in .gitignore about what api_key is.

Again, since you are only using fs for reading you might as well import the actual key itself.

Why do you need circumference calculation?

Rename your branch

again, if you are using ES6 destructure please put spaces between the variables and curly braces.

you still have single quotes here. => summoner.js page

sometimes in the then you do `( (s)` and sometimes you do `((s)` ; pick one.

make sure to convert all tabs to 2 spaces. 

place ; at the end of each statement. youre missing it in `app.get("/champs...`

File: src/LoLAPI/api.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+  Every function here is a call to LoL API only and should not do any complicated logi
+*/
+import fs from "fs";
+import request from "request"
+import polyfill from "babel-polyfill"
+
+let URL_GETSTAT = "https://na1.api.riotgames.com/lol/summoner/v3/summoners/by-name/"; // This need to be move out ot config file
+let URL_MATCHEES = "https://na1.api.riotgames.com/lol/match/v3/matchlists/by-account/"
+let URL_MATCHE = "https://na1.api.riotgames.com/lol/match/v3/timelines/by-match/"
```
Comment (cheriejw) [inline src/LoLAPI/api.js]: why is match spelled weird...

File: src/LoLAPI/api.js (hunk 4 above)
Comment (RichCherng) [reply src/LoLAPI/api.js:96]: I'm leaving it for now :D

**Comment by RichCherng** [comment]
Closed, Changing branch name

**Pull request closed without merging**